src/library/scala/collection/mutable/ArrayDeque.scala

```diff
@@ -483,9 +483,13 @@ class ArrayDeque[A] protected (
 
   override def sliding(window: Int, step: Int): Iterator[IterableCC[A]] = {
     require(window > 0 && step > 0, s"window=$window and step=$step, but both must be positive")
-    val lag = if (window > step) window - step else 0
-    if (length <= window) Iterator.single(slice(0, length))
-    else Iterator.range(start = 0, end = length - lag, step = step).map(i => slice(i, i + window))
+    val thisLength = length
+    if (thisLength == 0) Iterator.empty
+    else if (thisLength <= window) Iterator.single(slice(0, length))
+    else {
+      val lag = if (window > step) window - step else 0
+      Iterator.range(start = 0, end = thisLength - lag, step = step).map(i => slice(i, i + window))
+    }
```

Comment:
WDYT of not naming `thisLength`, and also making it clear that the alternative paths depend on length:
```
    length match {
      case 0 => Iterator.empty
      case n if n <= window => Iterator.single(slice(0, length))
      case n =>
        val lag = if (window > step) window - step else 0
        Iterator.range(start = 0, end = n - lag, step = step).map(i => slice(i, i + window))
    }
```
It would be interesting if the first case could read `case _ if isEmpty =>` in such a way as to avoid evaluating the scrutinee.

Reply:
:+1: done!